File: packages/sdk-snippets/src/targets/csharp/dotnet6/webhooks/fixtures/security-types.cs
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

// Your ReadMe secret
var secret = "my-readme-secret";

app.MapPost("/webhook", async context =>
{
  // Verify the request is legitimate and came from ReadMe.
  var signature = context.Request.Headers["readme-signature"];
  var body = await new StreamReader(context.Request.Body).ReadToEndAsync();

  try
  {
    ReadMe.Webhook.Verify(body, signature, secret);
  }
  catch (Exception e)
  {
    context.Response.StatusCode = StatusCodes.Status401Unauthorized;
    await context.Response.WriteAsJsonAsync(new
    {
      error = e.Message,
    });
    return;
  }

  // Fetch the user from the database and return their data for use with OpenAPI variables.
  // @todo Write your own query logic to fetch a user by `body["email"]`.

  await context.Response.WriteAsJsonAsync(new
  {
    // OAS Security variables
    keys = new[]
    {
      new
      {
        name = "api_key",
        apiKey = "default-api_key-key",
      },
      new
      {
        name = "http_basic",
        user = "user",
        pass = "pass",
      },
      new
      {
        name = "http_bearer",
        bearer = "default-http_bearer-key",
      },
      new
      {
        oauth2 = "default-oauth2-key",
      },
    }
  });
});

app.Run($"http://localhost:8000");

File: packages/sdk-snippets/src/targets/php/laravel/webhooks/fixtures/security-types.php
<?php

// Your ReadMe secret
$secret = 'my-readme-secret';

// Add this code into your `routes/web.php` file.
Route::post('/webhook', function (\Illuminate\Http\Request $request) use ($secret) {
    // Verify the request is legitimate and came from ReadMe.
    $signature = $request->headers->get('readme-signature', '');

    try {
        \ReadMe\Webhooks::verify($request->input(), $signature, $secret);
    } catch (\Exception $e) {
        // Handle invalid requests
        return response()->json([
            'error' => $e->getMessage()
        ], 401);
    }

    // Fetch the user from the database and return their data for use with OpenAPI variables.
    // $user = DB::table('users')->where('email', $request->input('email'))->limit(1)->get();
    return response()->json([
        // OAS Security variables
        'keys' => [
            [
                'name' => 'api_key',
                'apiKey' => 'default-api_key-key',
            ],
            [
                'name' => 'http_basic',
                'user' => 'user',
                'pass' => 'pass',
            ],
            [
                'name' => 'http_bearer',
                'bearer' => 'default-http_bearer-key',
            ],
            [
                'oauth2' => 'default-oauth2-key',
            ],
        ]
    ]);
});
